src/python/TaskWorker/Actions/PostJob.py

```diff
@@ -2607,8 +2621,58 @@ def set_dashboard_state(self, state, reason = None, exitCode = None):
                  }
         if reason:
             params['StatusValueReason'] = reason
-        ## Take exit code from job_fjr.<job_id>.<retry_id>.json and report final exit code to Dashboard.
-        ## Only taken if RetryJob think it is FATAL_ERROR.
+        monitoringExitCode(params, exitCode)
+        ## Unfortunately, Dashboard only has 1-second resolution; we must separate all
+        ## updates by at least 1s in order for it to get the correct ordering.
+        time.sleep(1)
+        msg += " Dashboard parameters: %s" % (str(params))
+        self.logger.info(msg)
+        DashboardAPI.apmonSend(params['MonitorID'], params['MonitorJobID'], params)
+        self.logger.info("====== Finished to prepare/send report to dashboard.")
+
+    ## = = = = = PostJob = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
+
+    def set_postJob_ClassAds(self, state, exitCode = None):
+        """
+        Update PostJobStatus and job exit-code among the job ClassAds for the monitoring script to update the Grafana dashboard.
+        """
+        if os.environ.get('TEST_POSTJOB_NO_STATUS_UPDATE', False):
+            return
+        msg = "postJob status: %s." % (state)
+        params = {'PostJobStatus': state}
+        monitoringExitCode(params, exitCode)
+        msg += " ClassAds values to set are: %s" % (str(params))
+        self.logger.info(msg)
+        backendurls = self.server.get(self.rest_uri_no_api + '/info', data = {'subresource': 'backendurls'})[0]['result'][0]
+        loc = HTCondorLocator.HTCondorLocator(backendurls)
+        address = None
+        schedd = None
+        try:
+            self.logger.debug("Getting schedd object")
+            schedd, address = loc.getScheddObjNew(task['tm_schedd'])
```

Comment:
@belforte, probably this part is not necessary as the postJob runs on the schedd currently.
If at some point we move it on another machine, then we need this code. What's your opinion?

Reply:
good point. Let me look at it carefull though. PostJos has a long history and I already tried and failed to clean it up one year ago. Don't break is our first priority.

Reply:
@lecriste  seems you are referring to new code that you are introducing, not to existing code.
I am all for reviewing design of how to add the new functionalities, better if before writing code !

No. There is no way that we run the PostJob in any other place then the schedd. It is part of the DAG. YOu can take some functionality out of PostJob and run it elsewhere, if useful. But "some PostJob" will always be there as long as we have a DAG.
Maybe I miss something ?